google-translate-workitem/src/main/java/org/jbpm/process/workitem/google/translate/TranslateWorkitemHandler.java

```
@@ -0,0 +1,110 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(TranslateWorkitemHandler.class);
+
+    private static final String RESULTS_TRANSLATION = "Translation";
+
+    private GoogleTranslateAuth googleTranslateAuth;
+    private Translate translationService;
+    private String apiKey;
+
+    public TranslateWorkitemHandler(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+        try {
+            String toStranslate = (String) workItem.getParameter("ToTranslate");
+            String sourceLang = (String) workItem.getParameter("SourceLang");
+            String targetLang = (String) workItem.getParameter("TargetLang");
+            Map<String, Object> results = new HashMap<String, Object>();
+
+            if (toStranslate == null || sourceLang == null || targetLang == null) {
+                logger.error("Missing translation text or source/target language.");
+                throw new IllegalArgumentException("Missing translation text or source/target language.");
+            }
+
+            translationService = googleTranslateAuth.getTranslationService(apiKey);
+
+            TranslateOption srcLangOption = TranslateOption.sourceLanguage(sourceLang);
+            TranslateOption targetLangOption = TranslateOption.sourceLanguage(targetLang);
+
+            Translation translation = translationService.translate(toStranslate,
+                                                                   srcLangOption,
+                                                                   targetLangOption);
+
+            if (translation != null && translation.getTranslatedText() != null) {
+                results.put(RESULTS_TRANSLATION,
+                            translation.getTranslatedText());
+            } else {
+                logger.error("Could not translate text: null");
+                results.put(RESULTS_TRANSLATION,
```

shouldn't it fail here instead of returning empty string... if we fail here people could use boundary error events to deal with such situations